### PR TITLE
Fixed scroll behavior and category tree bugs

### PR DESCRIPTION
**Pull request by gogo4ds**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1374
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1378
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1386

**Summary of the changes made**:
1. Upgraded to RouterProvider (https://reactrouter.com/en/main/upgrading/v6-data)
2. Fixed category tree behavior
   - used material UI TreeView component (https://mui.com/x/react-tree-view/)
   - items are not closing automatically
   - clicking on parent categories no request is made for contests, but just opens children (like in legacy judge)
4. Fixed scroll behavior with <ScrollRestoration /> (https://reactrouter.com/en/main/components/scroll-restoration)
5. Fixed document titles update logic

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.module.scss [hunk 1]
```diff
@@ -29,16 +29,46 @@
...
     padding: spacings.$sp-12 spacings.$sp-4;
     transition: background-color 0.2s ease-in-out;
 
+    &.darkCategoryItem {
+      border-color: colors.$dark-text-color;
+      color: colors.$dark-text-color;
+    }
+
+    &.mainTreeCategoryItem {
+      padding-left: 28px;
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.module.scss:69]: use spacings

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/home/HomeInfoPage.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    return (
+        <div className={styles.homePageStatisticWrapper}>
+            <Icon
+              size={IconSize.ExtraLarge}
+              Component={iconType}
+              color={themeColors.baseColor100}
+            />
+            <div className={styles.homePageStatisticTextWrapper}>
+                <div style={{ color: '#44a9f8' }}>{title}</div>
```
Comment (shsimeonova) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/home/HomeInfoPage.tsx:37]: Is this color compatible with the light theme?

**Review by shsimeonova — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.module.scss (hunk 1 above)
Comment (gogo4ds) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.module.scss:69]: We don't have the exact spacing for this case

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/home/HomeInfoPage.tsx (hunk 2 above)
Comment (gogo4ds) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/home/HomeInfoPage.tsx:37]: This is just moved file, I didn't introduce this color now..

**Review by shsimeonova — APPROVED** [review]

**Pull request merged**